src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -112,22 +112,20 @@ public Person toModelType() throws IllegalValueException {
         }
         final Name modelName = new Name(name);
 
-        if (company == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Company.class.getSimpleName()));
-        }
-        if (!Company.isValidCompany(company)) {
-            throw new IllegalValueException(Company.MESSAGE_CONSTRAINTS);
+        if (!(company == null)) {
+            if (!Company.isValidCompany(company)) {
+                throw new IllegalValueException(Company.MESSAGE_CONSTRAINTS);
+            }
         }
-        final Company modelCompany = new Company(company);
 
-        if (jobTitle == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
-                    JobTitle.class.getSimpleName()));
-        }
-        if (!JobTitle.isValidJobTitle(jobTitle)) {
-            throw new IllegalValueException(JobTitle.MESSAGE_CONSTRAINTS);
+        final Company modelCompany = company == null ? null : new Company(company);
```

Comment:
I think it's more common to do a null check as follows
```suggestion
        final Company modelCompany = company != null ? new Company(company) : null;
```
Same for job title